transbank/onepay/transaction.py

```diff
@@ -0,0 +1,122 @@
+
+class TransactionCreateResponse(TransbankRequestResponse):
+    signable_attributes = ['occ', 'external_unique_number', 'issued_at']
+
+    def __init__(self, occ, ott, signature, external_unique_number, issued_at, qr_code_as_base64):
+        self.occ = occ
+        self.ott = ott
+        self.signature = signature
+        self.external_unique_number = external_unique_number
+        self.issued_at = issued_at
+        self.qr_code_as_base64 = qr_code_as_base64
+
+class Transaction(object):
+
+    __SEND_TRANSACTION = "sendtransaction"
+    __COMMIT_TRANSACTION = "gettransactionnumber"
+    __TRANSACTION_BASE_PATH = '/ewallet-plugin-api-services/services/transactionservice/'
+
+    @classmethod
+    def create(cls, shopping_cart: ShoppingCart, channel = None, external_unique_number = None, options = None):
+
+        if (channel != None and channel == Channel.APP and onepay.app_scheme):
+            raise TransactionCreateError("You need to set an app_scheme if you want to use the APP channel")
+
+        if (channel != None and channel == Channel.MOBILE and onepay.callback_url):
+            raise TransactionCreateError("You need to set valid callback if you want to use the MOBILE channel")
+
+        if not hasattr(shopping_cart, 'items') or (hasattr(shopping_cart, 'items') and not shopping_cart.items):
+            raise ValueError("Shopping cart must not be null or empty")
+
+        path = cls.__TRANSACTION_BASE_PATH + cls.__SEND_TRANSACTION
+        api_base = onepay.integration_type.value.api_base
+
+        parsed_url = urlparse(api_base)
+        if parsed_url.scheme.lower() == "http":
+            conn = http.client.HTTPConnection(parsed_url.netloc)
+        else:
+            conn = http.client.HTTPSConnection(parsed_url.netloc)
+
+        external_unique_number_req = external_unique_number or datetime.utcnow().timestamp() * 1e3
+
+        req = TransactionCreateRequest(external_unique_number_req,
+              shopping_cart.total, shopping_cart.item_quantity,
+              calendar.timegm(datetime.utcnow().utctimetuple()), shopping_cart.items,
+              onepay.callback_url, channel.value , onepay.app_scheme, options)
+
+        try:
+            conn.request("POST", path, TransactionCreateRequestSchema().dumps(req).data)
+        except Exception:
+            raise TransactionCreateError("Could not obtain a response from the service")
+
+        data_response = conn.getresponse().read()
+        conn.close()
+
+        transaction_response = SendTransactionResponseSchema().loads(data_response.decode('utf-8')).data
+
+        if transaction_response['response_code'] != "OK":
+            raise TransactionCreateError("%s : %s" % (transaction_response['response_code'], transaction_response['description']))
+
+        result = TransactionCreateResponse(**transaction_response['result'])
+
+        if not sign.validate_create_response(result, (options or onepay).shared_secret, result.signature):
```

Comment:
So here this would be `if not sign.is_valid_signature(result, (options or onepay).shared_secret, result.signature)` and everything should work fine :) 

[I'm sure it won't, but I prefer to write code to explain a design change even if I risk writing code that doesn't work straight away. Explaining design changes in just english doesn't work for me]